tools/sense_studio/static/video_recording.js

```diff
@@ -0,0 +1,137 @@
+
+function increase(inputID) {
+    let element = document.getElementById(inputID);
+    let value = parseInt(element.value);
+
+    if (isNaN(value)) {
+        value = 1;
+    } else {
+        value++;
+    }
+
+    element.value = value;
+}
+
+
+function decrease(inputID, minValue) {
+    let element = document.getElementById(inputID);
+    let value = parseInt(element.value);
+
+    if (isNaN(value)) {
+        value = 1;
+    } else if (value > minValue) {
+        value--;
+    }
+
+    element.value = value;
+}
+
+
+function recordVideo(url) {
+    navigator.mediaDevices.getUserMedia({ audio: false, video: true })
+        .then(stream => setupRecording(stream, url));
+}
+
+
+function displayOverlay(text, mode) {
+    overlay = document.getElementById('textOverlay');
+    container = document.getElementById('videoContainer');
+
+    // Update overlay text
+    overlay.innerHTML = text;
+
+    // Update overlay visibility
+    if (text !== '') {
+        overlay.classList.remove('hidden');
+    } else {
+        overlay.classList.add('hidden');
+    }
+
+    // Update overlay color
+    if (mode === 'saved') {
+        overlay.classList.add('green');
+        overlay.classList.remove('red');
+    } else if (mode === 'error') {
+        overlay.classList.remove('green');
+        overlay.classList.add('red');
+    } else {
+        overlay.classList.remove('green');
+        overlay.classList.remove('red');
+    }
+
+    // Update container color
+    if (mode === 'recording') {
+        container.classList.add('red');
+        container.classList.add('inverted');
+    } else {
+        container.classList.remove('red');
+        container.classList.remove('inverted');
+    }
+}
+
+
+function setupRecording(stream, url) {
+    let player = document.getElementById('player');
+    player.srcObject = stream;
+
+    countdownDuration = parseInt(document.getElementById('countdown').value)
+    recordingDuration = parseInt(document.getElementById('duration').value)
+
+    // Show countdown
+    for (const seconds of Array(countdownDuration).keys()) {
+        const countdown = countdownDuration - seconds;
+        setTimeout(displayOverlay, seconds * 1000, 'Get Ready: ' + countdown, 'countdown');
+    }
+
+    // Start recording
+    setTimeout(startRecording, countdownDuration * 1000, stream, recordingDuration, url);
+};
+
+
+function startRecording(stream, recordingDuration, url) {
+    mediaRecorder = new MediaRecorder(stream, {mimeType: 'video/webm; codecs=vp8'});
```

Comment:
This is a global variable. I think here you're doing it on purpose. Better would be to wrap the media recorder logic in its own [js module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) and have the variable only exist in the module scope.

If it _really_ has to be global (which I think it shouldn't), I suggest we follow the convention of attaching it to the window object explicitly like so:

```
window.mediaRecorder = new MediaRecorder(...);
```

Reply:
I think it doesn't even need to be global, because I pass it over to the stopping function below.

Reply:
Ah, got it. So now it's a global variable, but you pass it to the stop function, and the parameter name (being the same) shadows the global variable. :exploding_head: 

In any case, what would you think about using proper modules? Not _required_, but a media recorder like this one could lend itself to starting with a proper code structure

Reply:
Wait, after my latest change there should be nothing global left.

Proper code structure sounds good though :grin: I'll definitely add that to the refactoring story.

Reply:
Oh yeah I commented before I had refreshed the code view, my comment was still for the previous version. It's fine now.